metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -169,6 +180,7 @@ class MetalsLanguageServer(
   private var doctor: Doctor = _
   var httpServer: Option[MetalsHttpServer] = None
   var treeView: TreeViewProvider = NoopTreeViewProvider
+  var worksheetProvider: WorksheetProvider = NoopWorksheetProvider
```

Comment:
Would it be possible to just use Option instead?

Reply:
That's a great question! I refactored to use `Option[WorksheetProvider]` and it came out much nicer